subprojects/provider/src/test/kotlin/org/gradle/kotlin/dsl/NamedContainersEvalTest.kt

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl
+
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.fixtures.TestWithTempFiles
+import org.gradle.kotlin.dsl.fixtures.eval
+import org.gradle.kotlin.dsl.fixtures.newProjectBuilderProject
+
+import org.hamcrest.CoreMatchers.hasItem
+import org.hamcrest.CoreMatchers.hasItems
+import org.hamcrest.MatcherAssert.assertThat
+
+import org.junit.Test
+
+
+class NamedContainersEvalTest : TestWithTempFiles() {
+
+    companion object {
+
+        val preExistingConfigurations = listOf(
+            "foo", "bar", "cabin", "castle"
+        )
+
+        val expectedConfigurationsExtendsFrom = listOf(
+            "bar" to "foo",
+            "cathedral" to "bazar",
+            "castle" to "cabin",
+            "hill" to "valley"
+        )
+    }
+
+    private
+    fun assertConfigurationsExtendsFrom(name: String, script: String, configuration: Project.() -> Unit = {}) {
+        newFolder(name).newProjectBuilderProject().run {
+
+            preExistingConfigurations.forEach { name ->
+                configurations.register(name)
+            }
+
+            configuration()
+
+            eval(script)
+
+            assertThat(
+                configurations.names.sorted(),
+                hasItems(
+                    *expectedConfigurationsExtendsFrom.flatMap {
+                        listOf(it.first, it.second)
+                    }.sorted().toTypedArray()
+                )
+            )
+            expectedConfigurationsExtendsFrom.forEach { (first, second) ->
+                assertThat(configurations[first].extendsFrom, hasItem(configurations[second]))
+            }
+        }
+    }
+
+    @Test
+    fun `monomorphic named domain object container api`() {
+
+        assertConfigurationsExtendsFrom("api", """
```

Comment:
:x: The name of this function made me think that the tested configurations should all extend from a configuration named `"api"`. How about renaming to something like `testConfigurationContainerVia("api", """...""")`?